packages/framework-provider-azure/src/library/events-store-adapter.ts

```diff
@@ -0,0 +1,140 @@
+import { BoosterConfig, EventEnvelope, NonPersistedEventEnvelope } from '@boostercloud/framework-types'
+import { CosmosClient, CreateOperationInput, JSONObject, OperationResponse, Response } from '@azure/cosmos'
+import { getLogger } from '@boostercloud/framework-common-helpers'
+import { partitionKeyForEvent } from './partition-keys'
+import { eventsStoreAttributes } from '../constants'
+
+type NonPersistedEventEnvelopePerPartitionKey = Record<string, Array<NonPersistedEventEnvelope>>
+
+const DEFAULT_CHUNK_SIZE = 100
+
+/**
+ * Limits: The Azure Cosmos DB request size limit constrains the size of the TransactionalBatch payload to not exceed 2 MB,
+ * and the maximum execution time is 5 seconds. There's a current limit of 100 operations per TransactionalBatch to ensure
+ * the performance is as expected and within SLAs.
+ *
+ * Errors: If there's a failure, the failed operation will have a status code of its corresponding error. All the other
+ * operations will have a 424 status code (failed dependency). The status code enables one to identify the cause of transaction failure.
+ *
+ * @param cosmosDb
+ * @param eventEnvelopes
+ * @param config
+ */
+export async function storeEvents(
+  cosmosDb: CosmosClient,
+  eventEnvelopes: Array<NonPersistedEventEnvelope>,
+  config: BoosterConfig
+): Promise<Array<EventEnvelope>> {
+  const logger = getLogger(config, 'store-events-adapter#storeEvents')
+  logger.debug('Storing EventEnvelopes with eventEnvelopes:', eventEnvelopes)
+  const envelopesWithCreatedAt: Array<EventEnvelope> = []
+
+  const eventsPerPartitionKey = eventEnvelopes.reduce(groupByPartitionKey, {})
+  for (const eventsPerPartitionKeyEntry of Object.entries(eventsPerPartitionKey)) {
+    const partitionKey = eventsPerPartitionKeyEntry[0]
+    const eventsInPartitionKey = eventsPerPartitionKeyEntry[1]
```

Comment:
You can simplify this with:
```suggestion
  for (const [partitionKey, eventsInPartitionKey] of Object.entries(eventsPerPartitionKey)) {
```

Reply:
Simpler and more elegant. Suggestion applied in commit 11bbb506.